src/main/java/org/openrewrite/analysis/search/UriCreatedWithHttpScheme.java

```diff
@@ -31,27 +35,30 @@ public class UriCreatedWithHttpScheme extends Recipe {
 
     private static final LocalFlowSpec<J.Literal, Expression> INSECURE_URI_CREATE = new LocalFlowSpec<J.Literal, Expression>() {
         @Override
-        public boolean isSource(J.Literal source, Cursor cursor) {
-            return source.getValue() != null && source.getValue().toString().startsWith("http://");
+        public boolean isSource(DataFlowNode srcNode) {
+            return srcNode
+                    .asExpr(Literal.class)
+                    .flatMap(Literal::getValue)
+                    .map(v -> v.toString().startsWith("http://"))
+                    .orElse(false);
         }
 
         @Override
-        public boolean isSink(Expression sink, Cursor cursor) {
-            J.MethodInvocation maybeMethodInvocation = cursor.firstEnclosing(J.MethodInvocation.class);
-            if (maybeMethodInvocation != null && URI_CREATE.matches(maybeMethodInvocation)) {
-                return maybeMethodInvocation.getArguments().contains(sink);
-            } else {
-                return false;
-            }
+        public boolean isSink(DataFlowNode sinkNode) {
+            return sinkNode
+                    .asExpr(MethodAccess.class)
+                    .map(ma -> ma.matches(URI_CREATE)) // TODO: getArguments().contains(sink)?
+                    .orElse(false);
```

Comment:
This isn't quite right, this should be something like:

```suggestion
            return URI_CREATE.advanced().isAnyArgument(sinkNode.getCursor());
```

I'm on my phone so this is mostly a guess on ensuring that I've got the right method names

Reply:
`URI_CREATE` is a `MethodMatcher`, not an `InvocationMatcher`, so it doesn't have those methods available. We could possibly change it to be an `InvocationMatcher` though.